### PR TITLE
refactor : add typehints to program module

### LINKED ISSUE #247: Improvement: Add typehints to Programs module
## Description

> Type hinting is a formal solution to statically indicate the type of a value within your Python code.

As a developer,
I need to add typehints to my code
to build and maintain a cleaner architecture and catch certain errors along the way.

## Mocks
https://github.com/anitab-org/bridge-in-tech-backend/blob/68d8ff9a8640cef1d3e53b2b0dd9b80e3d750fd7/app/api/dao/organization.py#L179-L182
Here's how to add typehints in above function.
```python
def get_organization(user_id: int, user_name: str):
    representative_additional_info = UserExtensionModel.find_by_user_id(user_id)
    ...
    ...
```
## Resources
Typehints cheatsheet - [Click Here](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

### Update [Required]
- [ ] Added correct typehints in the files related to Programs module.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 hour

### PR DESCRIPTION
**Pull request by Anmollenka**
### Description

Added typehints to the program modules

Fixes #247 

### Type of Change:

- Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged

**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by Anmollenka** [comment]
@vj-codes Can you please review this.

File: app/api/dao/program.py
Comment (mtreacy002) [inline app/api/dao/program.py]: @Anmollenka , why is this reimported again? It was already in the existing code line 15.

Comment (mtreacy002) [reply app/api/dao/program.py]: also, lint workflow failed. please run `black .` locally again 😉

**Comment by Anmollenka** [comment]
@mtreacy002 My bad while adding a new import statement it got pasted.

**Comment by Anmollenka** [comment]
Will make the required changes.

**Comment by Anmollenka** [comment]
@mtreacy002 Could you check if the lint workflow are failing again?

**Review by mtreacy002 — APPROVED** [review]
@Anmollenka , it looks good to me 🎉 🎉 🎉 . Now we just need one more reviewer to approve this PR 😉

**Review by isabelcosta — APPROVED** [review]
Thank you for contributing @Anmollenka!

**Comment by isabelcosta** [comment]
@mtreacy002 I approved it, do you think this is a "ready to merge"? I think it could be, if tests pass, this does not affect the program in runtime.
i know you are a little unavailable, so I'll merge :)

**Comment by mtreacy002** [comment]
Thank you @isabelcosta . Sorry about the delay in response 😊

**Pull request merged**
